### PR TITLE
Feature/update old modules

### PR DESCRIPTION
**Pull request by xadamxk**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,11 +4,14 @@
   "description": "Enhance your HF experience, a native browser extension.",
   "main": "src/HFX.js",
   "scripts": {
-    "lint": "./node_modules/.bin/eslint \"./src/**\" \"./tests/**\"",
+    "lint": "./node_modules/.bin/eslint \"./src/**\" \"./tests/**\" --fix",
```
Comment (xadamxk) [inline package.json:7]: This doesn't line up with the linting settings on my personal PC (which is also eslint), so I need to figure out why.

Comment (TyrantKingBen) [reply package.json:7]: Try setting `"root": true` in `.eslintrc.json` (located in the repos root directory) then running again.

File: package.json [hunk 2]
```diff
@@ -4,11 +4,14 @@
...
     "test": "jest",
     "config": "yarn && ./node_modules/.bin/gulp libs && ./node_modules/.bin/gulp build",
     "template": "node ./templates/index.js"
   },
+  "pre-commit": [
+    "lint"
```
Comment (TyrantKingBen) [inline package.json:13]: Maybe have it run the tests as well?

File: package.json (hunk 1 above)
Comment (xadamxk) [reply package.json:7]: Didn't seem to work. My local adds the space between function and it's parenthesis while the linter wants to remove the space:
```$(".post").each(function (index) {```

Unsure which is typically done.

File: src/features/global/CollapsePossiblyRelatedThreads.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ class CollapsePossiblyRelatedThreads extends Feature {
       section: Global,
       name: "Collapse Possibly Related Threads",
       default: false,
-      description: "Automatically collapse 'Possibly Related Threads' results.\n\nClick the toggle image to view results"
+      description: "Automatically collapse 'Possibly Related Threads' results.\n\nClick the toggle image to view results."
```
Comment (xadamxk) [inline src/features/global/CollapsePossiblyRelatedThreads.js:11]: I deleted this file. Did you pull my branch before you made these changes?

Comment (TyrantKingBen) [reply src/features/global/CollapsePossiblyRelatedThreads.js:11]: You forgot to commit the deletion then.

File: package.json [hunk 4]
```diff
@@ -43,7 +48,8 @@
     "gulplog": "^1.0.0",
     "jest": "^24.9.0",
     "mustache": "^3.1.0",
+    "pre-commit": "^1.2.2",
```
Comment (TyrantKingBen) [inline package.json:51]: This breaks committing via GitHub Desktop for me. I believe because the precommit hook it installs needs to be ran in a terminal.

Comment (xadamxk) [reply package.json:51]: Well, that is problematic then. 
I'll try to find a workaround.

Comment (TyrantKingBen) [reply package.json]: Switched it to using `husky`. Works great from CLI and GitHub Desktop (and appears to be maintained).

**Pull request merged**